### DISCUSSION TITLE
Using CUE to add multiple input files to a list?

### DISCUSSION THREAD

**Discussion by kimfy** [cue-lang/cue#2582] [Q&A] [answered]
I am trying to find out if merging multiple JSON files which only have one object in it to a list is possible with CUE. This seems like the simplest thing but I am not understanding correctly.

Consider the following file structure:
```bash
../
./
merge.cue
input1.json
input2.json
input3.json
```

In merge.cue I have a schema
```cue
#Foo: {
  oof: string
  rab: string
}

// My non-working output is
foos: [...#Foo]
```

I want the output to be
```json
{
  "foos": [
    { "oof": "", "rab": "" },
    # and so on
  ]
```

I have been running with the command `cue eval merge.cue *.json --out json`

Is this possible - and if so, can someone point me in the right direction.

Please, and thanks. 👐🏻

**Comment by myitcv** [comment, marked as answer]
Hi @kimfy - welcome to the CUE project. 

Great question. Here is an example of what I think you're after:

```
exec cue export --with-context -l '_inputs:' -l filename input1.json input2.json input3.json input4.json input5.json x.cue
cmp stdout stdout.golden

-- input1.json --
{ "oof": "test1" }
-- input2.json --
{ "oof": "test2" }
-- input3.json --
{ "oof": "test3" }
-- input4.json --
{ "oof": "test4" }
-- input5.json --
{ "oof": "test5" }
-- x.cue --
package x

#Foo: {
	oof?: string
	rab?: string
}

_inputs: [string]: #Foo
foos: [
	for _, v in _inputs {v},
]
-- stdout.golden --
{
    "foos": [
        {
            "oof": "test1"
        },
        {
            "oof": "test2"
        },
        {
            "oof": "test3"
        },
        {
            "oof": "test4"
        },
        {
            "oof": "test5"
        }
    ]
}
```

`cue help flags` will show you more information about the `--with-context` flag. 

Notice the use of the `-l` flag, which places the non-CUE data at a particular path.

**Comment by kimfy** [comment]
Hi @myitcv - thanks for that. I don't think I'd ever find that out by myself. I'm trying to think of ways I could contribute to the documentation, hopefully in a couple of months when I am more familiar with CUE. Wishing you the best!

Kim
